croncat/src/modules/tasks.rs

```diff
@@ -50,6 +49,34 @@ pub struct BatchQueryMsg {
     pub batch_query: BatchQuery,
 }
 
+use std::{error::Error as StdError, fmt};
+
+#[derive(Debug)]
+pub struct CustomError {
+    inner: Box<dyn StdError + Send + Sync>,
+}
+
+impl fmt::Display for CustomError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.inner)
+    }
+}
+
+impl StdError for CustomError {
+    fn source(&self) -> Option<&(dyn StdError + 'static)> {
+        self.inner.source()
+    }
+}
+
+// impl From<ErrReport> for CustomError {
+//     fn from(err: ErrReport) -> CustomError {
+//         // Perform the custom conversion here
+//         CustomError {
+//             inner: Box::new(err),
+//         }
+//     }
+// }
+
```

Comment:
Checkout [snafu](https://docs.rs/snafu/latest/snafu/#struct-style) for something like this!